Source/WebCore/platform/graphics/coretext/FontCoreText.cpp

```diff
@@ -221,6 +221,25 @@ void Font::platformInit()
             xHeight = verticalRightOrientationFont().fontMetrics().xHeight();
     }
 
+    if (CTFontGetSymbolicTraits(m_platformData.font()) & kCTFontTraitColorGlyphs) {
+#if HAVE(CTFONTCOPYCOLORGLYPHCOVERAGE)
+        // The reason this is guarded with both a preprocessor define and soft linking is that
+        // we want to get rid of the soft linking soon,
+        // once people have a chance to update to an SDK that includes it.
+        // At that point, only the preprocessor define will remain.
+        if (PAL::canLoad_CoreText_CTFontCopyColorGlyphCoverage()) {
+            if (auto cfBitVector = adoptCF(PAL::softLink_CoreText_CTFontCopyColorGlyphCoverage(m_platformData.font())))
+                m_emojiType = SomeEmojiGlyphs { BitVector(cfBitVector.get()) };
+            else
+                m_emojiType = NoEmojiGlyphs { };
+        } else
+#endif
+        {
+            m_emojiType = AllEmojiGlyphs { };
```

Comment:
It's not clear to me what the result of using AllEmojiGlyphs is here, when CTFontCopyColorGlyphCoverage is not available (like on downlevels). Maybe no change for current behavior, since we don't implement font-variant-emoji (and so won't use ResolveEmojiPolicy::RequireText)?

Reply:
The idea is, unless you have glyph-by-glyph information, the font itself has to be classified as either a "color font" or a "non-color font." Color fonts can honor requests for emoji-style characters. So we say, if `CTFontGetSymbolicTraits(m_platformData.font()) & kCTFontTraitColorGlyphs` is true, act as if every glyph in the font is an emoji-style glyph, because the font is a color font.